src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -222,8 +225,13 @@ def construct_payload(self):
 
             # Check if updating existing container
             updating_existing_container = False
+            if self.get_argument_force_single_container_updates():
```

Comment:
This logic is for preview feature and not related to any GA feature, should not be added in `ContainerAppUpdateDecorator`.
If you need to reuse this part of logic for update_containers, please split it as follow example then in `ContainerAppPreviewUpdateDecorator` over write it.:
```

def should_update_existing_container(self, curent_container_name=None):
  return curent_container_name.lower() == self.get_argument_container_name().lower() or self.get_argument_force_single_container_updates()

def set_up_updating_existing_container():
    # using should_update_existing_container
    if self.should_update_existing_container():


```

Reply:
@Greedygre could you please elaborate on how splitting this would ensure the change is only introduced in the `ContainerAppPreviewUpdateDecorator`? 

Since the payload for `self.new_containerapp`  is populated in `ContainerAppUpdateDecorator` , there would be no way to properly determine cloud build scenario to remove all the existing containers in `ContainerAppPreviewUpdateDecorator`

Reply:
Here is an example. You can organize the code to make it more elegant.
In ContainerAppUpdateDecorator:
![image](https://github.com/Azure/azure-cli-extensions/assets/46143499/96a85b9d-bf21-4725-98dd-12ffdc41f0f8)

In ContainerAppPreviewUpdateDecorator, overwrite key logic that needs to be modified in the subclass

```
    def _need_update_container(self):
        return self.get_argument_image() or self.get_argument_force_single_container_updates() or self.get_argument_container_name() or self.get_argument_set_env_vars() is not None or self.get_argument_remove_env_vars() is not None or self.get_argument_replace_env_vars() is not None or self.get_argument_remove_all_env_vars() or self.get_argument_cpu() or self.get_argument_memory() or self.get_argument_startup_command() is not None or self.get_argument_args() is not None or self.get_argument_secret_volume_mount() is not None

    def set_up_containers(self):
        if self.get_argument_force_single_container_updates():
            # Remove n-1 containers where n is the number of containers in the containerapp and replace remaining container with the new container
            # Fails if all containers are removed
            while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
                self.new_containerapp["properties"]["template"]["containers"].pop()
        super().set_up_containers()

    def should_update_existing_container(self, c) -> bool:
        return c["name"].lower() == self.get_argument_container_name().lower() or self.get_argument_force_single_container_updates()
```

Reply:
@Greedygre Is there a way to make sure the preview code is executed first? The above proposed changes fails the test because the code in `ContainerAppPreviewUpdateDecorator` did not execute first before the code in `ContainerAppUpdateDecorator`.